sdk/servicebus/azure-messaging-servicebus/src/main/java/com/azure/messaging/servicebus/ServiceBusMessage.java

```diff
@@ -141,17 +157,34 @@ public AmqpAnnotatedMessage getAmqpAnnotatedMessage() {
      * @return A byte array representing the data.
      */
     public byte[] getBody() {
-        final AmqpBodyType type = amqpAnnotatedMessage.getBody().getBodyType();
-        switch (type) {
+        return getBodyAsBinaryData().toBytes();
+    }
+
+    /**
+     * Gets the actual payload/data wrapped by the {@link ServiceBusMessage}.
+     *
+     * <p>
+     * If the means for deserializing the raw data is not apparent to consumers, a common technique is to make use of
+     * {@link #getApplicationProperties()} when creating the event, to associate serialization hints as an aid to
+     * consumers who wish to deserialize the binary data.
+     * </p>
+     *
+     * @return A {@link BinaryData} representing the data.
+     */
+    public BinaryData getBodyAsBinaryData() {
+        final AmqpBodyType bodyType = amqpAnnotatedMessage.getBody().getBodyType();
+        switch (bodyType) {
             case DATA:
-                return Arrays.copyOf(binaryData, binaryData.length);
+                return ((AmqpDataBody) amqpAnnotatedMessage.getBody()).getData().stream().findFirst().get();
+
             case SEQUENCE:
             case VALUE:
-                throw logger.logExceptionAsError(new UnsupportedOperationException("Not supported AmqpBodyType: "
-                    + type.toString()));
+                throw logger.logExceptionAsError(new UnsupportedOperationException("Body type not supported yet "
+                    + bodyType.toString()));
             default:
-                throw logger.logExceptionAsError(new IllegalArgumentException("Unknown AmqpBodyType: "
-                    + type.toString()));
+                logger.warning("Invalid body type {}.", bodyType);
+                throw logger.logExceptionAsError(new IllegalStateException("Body type not valid "
```

Comment:
Same question about why we are logging both a warning and throwing a logged error.